src/gridcoin.cpp

```diff
@@ -6,14 +6,220 @@ namespace
+        uint256S("fbbfd66c63b8d7a07472f29c612a14906c9ff3b2db4e10a6453ccea68d50f035"), //P1286867 (beaconalt)
+        uint256S("a6417db646615f57c2fcca0df5369d442ea98d0c6f747b82d30ab58cd8bcb6c5"), //P1324021 (beaconalt)
+        uint256S("bfb6d8f1dbe196b1796df84fa2472dbe785972406fee728ab5d4bd9f229460fd"), //P1337709 (beaconalt)
+        uint256S("f94fd82db5dbb0a8dd6145573ad3ff0759e74cd26262a0ec5e8609c36bbdc742"), //P1350028 (beaconalt)
+        uint256S("1fe190274629969ff626cd0cd446b3159b947a47b2cdf783e90f2fc40da2208a"), //P1351825 (beaconalt)
+        uint256S("e4ce4a928e8d42db26e16eb657f8e53f4be401ba480de01af70d5ae1228677d1"), //P1368309 (beaconalt)
+        uint256S("ab06a403875c6c97c6ce3e68fffc19d9dbf3be64324c64d9dbddc4958b350fab"), //P1441206 (beaconalt)
+        uint256S("ede0a3eda3d53546e1e7b340b6fad7b428aecdd85bcf7534def8365e1b89828c"), //P1441467 (beaconalt)
+        uint256S("bff64c0bf316c442f1f9e317589a6eb38ffce6623c03867e0e48853d15434665"), //P1564089 (beaconalt)
+        uint256S("b67f1bf92dda31c833cb37f8fd0cd0eb4068f9bfb023af0c86600e922ef99082"), //P1565898 (beaconalt)
+        uint256S("a9eb4e9d7b7309805e0dc30ad88fd81edbfd632b2a978f20c8f3a7454d42a288"), //P1774129 (beaconalt)
+        uint256S("520e1efde69a4fb62341533aaef51f8008f17b16555bc738504941910238af43"), //P1862976 (beaconalt contract)
+        uint256S("3cb0d638480886a43b0014e5e3ec1cd83d5d987d52913eff0c88139dabdfc605"), //P1875186 (beaconalt contract)
+        uint256S("e64d0e3b1885586ce97bb607b650a952559811ae3473581f23c80c2386a68ccf"), //P1883817 (beaconalt contract)
+        uint256S("6f3d843eb150afb98751b0bf1d25286f53d86bf9883822bdd0f6e91f3db9ddfd"), //P1884851 (beaconalt contract)
+        uint256S("be248382c451f32a485d163098e780c2cbd78f70effafc86d632c2554a081912"), //P1885013 (beaconalt contract)
+        uint256S("6058a61b27ae4d060712c454ba8b7436c72d3858164dada274c59b76ebc2d814"), //P1888603 (beaconalt contract)
+        uint256S("dec93cbf000d0c9cb64ecf528db6f6c005c4e6133380de935d95c4c1d97a3bde"), //P1888869 (beaconalt contract)
+        uint256S("25c164087ab2dc1eb96ed6589d43568bbb17a22fb987c9069053b53fb9548418"), //P1889534 (beaconalt contract)
+        uint256S("52f706715f7c90b202b39705eb410f9e3bbf97c47e2cfc130b95954eb767d456"), //P1890348 (beaconalt contract)
+        uint256S("b551bb5242c7ad5bdfe4336c26bba9974284abd81b9b39ac4e7044300a47eeb3"), //P1890742 (beaconalt contract)
+        uint256S("c0f4a662f41cf3a9aeefc9818344e94b0c615e857293306be5b48e7d42dc2362"), //P1890827 (beaconalt contract)
+        uint256S("44e60e6e6bb3414029b4ee4aa9410d3ef089a0dbca645a1d3210aecb3f227a4c"), //P1891639 (beaconalt contract)
+        uint256S("89da5f60b6dd1b1560e595d79ecccdb4ec18e27370b2d396d69ebc979930cb43"), //P1895527 (beaconalt contract)
+        uint256S("c34642b45b839c9aedb6173e8401c947e47347b90fbc210e3be164c0971033b2"), //P1896499 (beaconalt contract)
+        uint256S("c4efca7e9d5cf292f26476e4e8bd597b6350ff58ef47f34084cc8321a1f6d1c1"), //P1896921 (beaconalt contract)
```

Comment:
That's true. In practice, only nodes running version <= 4.0.1.0 will cause this problem, and only if one of these nodes loads a backup config file with an old beacon key that is still newer than 6 months old and manages to stake a block. 

I can put the workaround back in, but I'm not sure it's worth it.

Reply:
Let's talk about what happens if someone does that. The single node forks off?

Reply:
If that is all that happens, I am fine with it. Exchanges will not be posting beacons anyway, so we don't have to worry about them being on ancient wallet versions. I seem to remember that one of the pools was using ancient wallet versions, although it may have been greater than 4.0.1.0, and IIRC may have been upgraded already. Perhaps @barton2526 can remind of the details. He has a good memory for that sort of thing.

Reply:
Wait.... according to the exception list it is happening more often than you think! 1896921 - 1862976 + 1 = 33946 from the exception list above. This is 15 times in 33946 blocks or about once every 2 days.

I think we run the risk of a major fork. Think about the dynamics... We pin the mandatory height two weeks in the future, people start to upgrade, and then within a couple of days into the transition, one of these comes along, and forks off the *new code* nodes before the mandatory takes place, and before the population of new nodes reaches enough of a majority to force the old nodes into the lesser trust fork rather than the new nodes. The result of this dynamically will be all new nodes forked off and the chain continuing under the old rules... i.e. a failed transition.

I think we have to put the alt beacon structure back in. We can safely retire it and put this exception list in place the first leisure release after the mandatory.